### PR TITLE
#161997901 ability to attach tag to article 

### PR DESCRIPTION
**Pull request by koiic**
#### Description of Task to be completed?
- fetch tags by tag name
- display existing tags
- select tags
- attach tags to article
#### What are the relevant pivotal tracker stories?
[#161997901](https://www.pivotaltracker.com/story/show/161997901, "attach tags to an article")

#### Screenshots (if appropriate)
screen shot 2018-11-21 at 11 33 52 am

### CODE REVIEW

File: src/action/tag/tag.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    },
+  };
+
+  return http.post(`${BASE_URL}/alltags`, options).then((response) => {
+    dispatch(fetchTagSuccess(response.data))
+  })
+    .catch((err) => {
+      dispatch(fetchTagFailure(err.response))
+    });
+}
```
Comment (Mrshinezee) [inline src/action/tag/tag.js]: Please add a new line

**Review by Mrshinezee — CHANGES_REQUESTED** [review]

File: src/action/tag/tag.js (hunk 1 above)
Comment (koiic) [reply src/action/tag/tag.js]: Alright, Thanks i'll fix it now

**Review by Mrshinezee — APPROVED** [review]

File: src/components/login/LoginForm.jsx [hunk 2]
```diff
@@ -1,9 +1,9 @@
 // react libraries
 import React, { Component, Fragment } from 'react';
+import Loader from 'react-loader';
```
Comment (e0works) [inline src/components/login/LoginForm.jsx]: this should be under third-party libraries
```js
https://lotus-ah-frontend.herokuapp.com
```

File: src/components/reusables/userDetail/UserDetail.jsx [hunk 3]
```diff
@@ -1,37 +1,44 @@
 // react libraries
-import React from 'react'
+import React from 'react';
+import { Link } from 'react-router-dom';
```
Comment (e0works) [inline src/components/reusables/userDetail/UserDetail.jsx]: this should be under `third-party libraries`
```js
import { Link } from 'react-router-dom';
```

**Review by e0works — CHANGES_REQUESTED** [review]

File: src/containers/tag/SearchTag.jsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+          }
+        </div>
+      </div>
+    );
+  }
+}
+const mapDispatchToProps = dispatch => ({
+  fetchSingleTag : tagName => dispatch(fetchSingleTag(tagName)),
+})
+const mapStateToProps = (state = {}) => ({
```
Comment (KaiserPhemi) [inline src/containers/tag/SearchTag.jsx]: #### Critical
Why pass empty state here?

File: src/components/article/viewArticle/ViewArticlePage.jsx [hunk 5]
```diff
@@ -59,15 +59,18 @@ class ViewArticle extends Component {
     const {
       title,
       imageUrl,
-      body
+      body,
+      userId,
+      slug
     } = this.props.publishedArticle.Articles;
     const shareUrl = `https://lotus-ah-staging.herokuapp.com/api/v1${this.props.location.pathname}`;
+    // const articleOwner = this.props.publishedArticle.Articles.user.id;
```
Comment (KaiserPhemi) [inline src/components/article/viewArticle/ViewArticlePage.jsx]: #### Nitpick
Remove unneeded lines of code

**Review by KaiserPhemi — CHANGES_REQUESTED** [review]

File: src/containers/tag/SearchTag.jsx (hunk 4 above)
Comment (koiic) [reply src/containers/tag/SearchTag.jsx]: I will import and pass in the initialState

**Review by KaiserPhemi — APPROVED** [review]

**Pull request merged**